src/Models/CartRequirement.php

```diff
@@ -113,13 +113,13 @@ public function quantityIsEligible(CartItem $cartItem): array
 
         $firstRequirementsQuantityError = $this->catchQuantityMismatch($cartItem, $nonWhiteGloveProductVariantsInCart, $this->first_set_of_required_shopify_product_ids, );
 
-        if (! is_null($firstRequirementsQuantityError)) {
+        if (! reset($firstRequirementsQuantityError)) {
```

Comment:
This "smells" a little to me. It seems like you're trying to get the first element of the array, which should be a `boolean`, right? If that's the case I would personally opt for the more explicit call like `$firstRequirementsQuantityError[0]`. 

Using the `reset()` function would, to me, indicate more that you're looping through an array and at some point you need to stop and go back to the first element. The [PHP documentation](https://www.php.net/manual/en/function.reset.php) describes it like so:
> reset() rewinds array's internal pointer to the first element and returns the value of the first array element.

The word "rewinds" is where there's a disconnect for me with how the `reset` function is used here. We don't rewind it, we just need the first element.

One other thing to note is that the return type for the `catchQuantityMismatch` method says it can either be an `array|null` but it looks like it will always return an `array`? If not then you also need to take that into account when you do your check as both `$firstRequirementsQuantityError[0]` and `reset($firstRequirementsQuantityError)` will throw an exception if `$firstRequirementsQuantityError` is `null`.

Reply:
Great point of view @tonning, let me fix this. Thanks.

Reply:
I would also remove the `null` return type since the `catchQuantityMismatch` cannot return just a `null`... It will by default return an array `[true, null]`